src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -914,35 +1050,51 @@ bool HMICapabilitiesImpl::load_capabilities_from_file() {
 
     const bool result = reader_->parse(
         json_string.c_str(), json_string.c_str() + json_len, &root_json, &err);
+
     if (!result) {
       LOG4CXX_DEBUG(logger_, "Json parsing fails: " << err);
       return false;
     }
+
+    JsonCapabilitiesGetter json_root_getter(root_json, root_json_override);
     // UI
-    if (check_existing_json_member(root_json, "UI")) {
-      Json::Value ui = root_json.get("UI", Json::Value::null);
 
-      if (check_existing_json_member(ui, "language")) {
-        const std::string lang = ui.get("language", "EN-US").asString();
+    if (!json_root_getter.GetInterfaceJsonMember(hmi_interface::ui).isNull()) {
+      auto ui_main_node = json_root_getter.GetMainJsonMember(hmi_interface::ui);
+      auto ui_override_node =
```

Comment:
@LitvinenkoIra I suggest to name these variables accordingly to their meaning, for example:
`ui_default_node` and `ui_cache_node`